src/sinks/util/buffer/loki.rs

```diff
@@ -80,6 +124,41 @@ impl Batch for LokiBuffer {
     }
 
     fn push(&mut self, mut item: Self::Input) -> PushResult<Self::Input> {
+        // We must sort the stream labels to ensure they hash to
+        // the same stream if the label set matches.
+        item.labels.sort_unstable();
```

Comment:
This is something I had in mind too, but I would do it in another PR.

The keys never change during runtime so if we sort them (or use BTreeMap) at `LokiSink` it will stay sorted after template rendering. I think `BTreeMap` is kinda heavy in this situation, because there is no need to access value by key, therefore I'm hesitant to use it.